cmd/bee-file/main.go

```diff
@@ -0,0 +1,255 @@
+	// process the reference to retrieve
+	addr, err := swarm.ParseHexAddress(args[0])
+	if err != nil {
+		return err
+	}
+
+	// initialize interface with HTTP API
+	store := cmdfile.NewApiStore(host, port, ssl)
+
+	buf := bytes.NewBuffer(nil)
+	writeCloser := cmdfile.NopWriteCloser(buf)
+	limitBuf := cmdfile.NewLimitWriteCloser(writeCloser, limitMetadataLength)
+	j := joiner.NewSimpleJoiner(store)
+	err = cmdfile.JoinReadAll(j, addr, limitBuf)
+	if err != nil {
+		return err
+	}
+	e := &entry.Entry{}
+	err = e.UnmarshalBinary(buf.Bytes())
+	if err != nil {
+		return err
+	}
+
+	buf = bytes.NewBuffer(nil)
+	err = cmdfile.JoinReadAll(j, e.Metadata(), buf)
+	if err != nil {
+		return err
+	}
+
+	// retrieve metadata
+	metaData := &entry.Metadata{}
+	err = json.Unmarshal(buf.Bytes(), metaData)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("Filename: %s", metaData.Filename)
+	logger.Debugf("MIME-type: %s", metaData.MimeType)
+
+	if outDir == "" {
+		outDir = "."
+	} else {
+		err := os.MkdirAll(outDir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+	}
+	outFilePath := filepath.Join(outDir, metaData.Filename)
+
+	// create output dir if not exist
+	if outDir != "." {
+		err := os.MkdirAll(outDir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+	}
+
+	// protect any existing file unless explicitly told not to
+	outFileFlags := os.O_CREATE | os.O_WRONLY
+	if outFileForce {
+		outFileFlags |= os.O_TRUNC
+	} else {
+		outFileFlags |= os.O_EXCL
+	}
+
+	// open the file
+	outFile, err := os.OpenFile(outFilePath, outFileFlags, 0o666) // skipcq: GSC-G302
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+
+	err = cmdfile.JoinReadAll(j, e.Reference(), outFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// putEntry creates a new file entry with the given reference.
+func putEntry(cmd *cobra.Command, args []string) (err error) {
+	// process the reference to retrieve
+	addr, err := swarm.ParseHexAddress(args[0])
+	if err != nil {
+		return err
+	}
+	// add the fsStore and/or apiStore, depending on flags
+	stores := cmdfile.NewTeeStore()
+	if outDir != "" {
+		err := os.MkdirAll(outDir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+		store := cmdfile.NewFsStore(outDir)
+		stores.Add(store)
+	}
+	if useHttp {
+		store := cmdfile.NewApiStore(host, port, ssl)
+		stores.Add(store)
+	}
+
+	// create metadata object, with defaults for missing values
+	if filename == "" {
+		filename = args[0]
+	}
+	if mimeType == "" {
+		mimeType = defaultMimeType
+	}
+	metadata := entry.NewMetadata(filename)
+	metadata.MimeType = mimeType
+
+	// serialize metadata and send it to splitter
+	metadataBytes, err := json.Marshal(metadata)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("metadata contents: %s", metadataBytes)
+
+	// set up splitter to process the metadata
+	s := splitter.NewSimpleSplitter(stores)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
i wonder if we need the cancel func at all. i cant see this being used except in the defer.